source/extensions/health_checkers/thrift/thrift.cc

```diff
@@ -0,0 +1,119 @@
+#include "source/extensions/health_checkers/thrift/thrift.h"
+
+#include "source/extensions/filters/network/thrift_proxy/thrift.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HealthCheckers {
+namespace ThriftHealthChecker {
+
+namespace {
+
+// Helper functions to get the correct hostname for an L7 health check.
+const std::string& getHostname(const Upstream::HostSharedPtr& host,
+                               const Upstream::ClusterInfoConstSharedPtr& cluster) {
+  return host->hostnameForHealthChecks().empty() ? cluster->name()
+                                                 : host->hostnameForHealthChecks();
+}
+
+} // namespace
+
+ThriftHealthChecker::ThriftHealthChecker(
+    const Upstream::Cluster& cluster, const envoy::config::core::v3::HealthCheck& config,
+    const envoy::extensions::health_checkers::thrift::v3::Thrift& thrift_config,
+    Event::Dispatcher& dispatcher, Runtime::Loader& runtime,
+    Upstream::HealthCheckEventLoggerPtr&& event_logger, Api::Api& api,
+    ClientFactory& client_factory)
+    : HealthCheckerImplBase(cluster, config, dispatcher, runtime, api.randomGenerator(),
+                            std::move(event_logger)),
+      method_name_(thrift_config.method_name()),
+      transport_(ProtoUtils::getTransportType(thrift_config.transport())),
+      protocol_(ProtoUtils::getProtocolType(thrift_config.protocol())),
+      client_factory_(client_factory) {
+  if (transport_ == TransportType::Auto || protocol_ == ProtocolType::Auto ||
+      protocol_ == ProtocolType::Twitter) {
+    throw EnvoyException(
+        fmt::format("Invalid thrift health check configuration: {}", thrift_config.DebugString()));
```

Comment:
nit: should the error be more specific about what's invalid? Something like "unsupported transport or protocol in thrift health check configuration"?

Reply:
Good point. Will address it in separating PR if api is happy to move forward. Thanks!

Reply:
I'll address this now since matt requested review